src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1392,6 +1392,20 @@ public static function getServiceConditionals($value)
     }
 
     /**
+     * Computes a reasonably unique hash of a value.
+     *
+     * @param mixed $value A serializable value
+     *
+     * @return string
+     */
+    public static function hash($value)
+    {
+        $hash = substr(base64_encode(hash('sha256', serialize($value), true)), 0, 7);
+
+        return str_replace(array('/', '+'), array('.', '_'), strtolower($hash));
```

Comment:
That's because ids are lowercased anyway, and having them lowercase already prevents filling the internal "normalizedIds" property of containers. Could be removed on master.